braid/step.c

```diff
@@ -40,21 +41,113 @@ _braid_Step(braid_Core         core,
    braid_StepStatus   status   = (braid_StepStatus)core;
    braid_Int          nrefine  = _braid_CoreElt(core, nrefine);
    braid_Int          gupper   = _braid_CoreElt(core, gupper);
+   braid_Int          nlevels  = _braid_CoreElt(core, nlevels);
+   braid_Int          cfactor  = _braid_GridElt(grids[level], cfactor);
    braid_Int          ilower   = _braid_GridElt(grids[level], ilower);
    braid_Real        *ta       = _braid_GridElt(grids[level], ta);
    braid_BaseVector  *fa       = _braid_GridElt(grids[level], fa);
 
-   braid_Int        ii;
-
-   ii = index-ilower;
-   _braid_StepStatusInit(ta[ii-1], ta[ii], index-1, tol, iter, level, nrefine, gupper, status);
+   braid_Int ii = index-ilower;
 
    /* If ustop is set to NULL, use a default approach for setting it */
    if (ustop == NULL)
    {
       _braid_GetUInit(core, level, index, u, &ustop);
    }
 
+   if (_braid_CoreElt(core, delta_correct) && iter >= _braid_CoreElt(core, delta_defer_iter) && level >= _braid_CoreElt(core, delta_defer_lvl))
```

Comment:
Here's this logic again about whether delta_correct is enabled ... maybe a macro would make this more readable (and maintainable)